internal/service/networkmanager/vpn_attachment.go

```diff
@@ -0,0 +1,323 @@
+	}
+
+	return nil
+}
+
+func resourceVpnAttachmentUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).NetworkManagerConn
+
+	if d.HasChange("tags_all") {
+		o, n := d.GetChange("tags_all")
+
+		if err := UpdateTags(conn, d.Get("arn").(string), o, n); err != nil {
+			return diag.FromErr(fmt.Errorf("error updating Network Manager VPN Attachment (%s) tags: %s", d.Get("arn").(string), err))
+		}
+	}
+
+	return resourceVpnAttachmentRead(ctx, d, meta)
+}
+
+func resourceVpnAttachmentDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).NetworkManagerConn
+
+	// If ResourceAttachmentAccepter is used, then VPN Attachment state
+	// is never updated from StatePendingAttachmentAcceptance and the delete fails
+	output, sErr := FindVpnAttachmentByID(ctx, conn, d.Id())
+	if tfawserr.ErrCodeEquals(sErr, networkmanager.ErrCodeResourceNotFoundException) {
+		return nil
+	}
+
+	if sErr != nil {
+		return diag.Errorf("deleting Network Manager VPN Attachment (%s): %s", d.Id(), sErr)
+	}
+
+	d.Set("state", output.Attachment.State)
+
+	if state := d.Get("state").(string); state == networkmanager.AttachmentStatePendingAttachmentAcceptance || state == networkmanager.AttachmentStatePendingTagAcceptance {
+		return diag.Errorf("cannot delete Network Manager VPN Attachment (%s) in %s state", d.Id(), state)
+	}
+
+	log.Printf("[DEBUG] Deleting Network Manager VPN Attachment: %s", d.Id())
+	_, err := conn.DeleteAttachmentWithContext(ctx, &networkmanager.DeleteAttachmentInput{
+		AttachmentId: aws.String(d.Id()),
+	})
+
+	if tfawserr.ErrCodeEquals(err, networkmanager.ErrCodeResourceNotFoundException) {
+		return nil
+	}
+
+	if err != nil {
+		return diag.Errorf("deleting Network Manager VPN Attachment (%s): %s", d.Id(), err)
+	}
+
+	if _, err := waitVpnAttachmentDeleted(ctx, conn, d.Id(), d.Timeout(schema.TimeoutDelete)); err != nil {
+		return diag.Errorf("waiting for Network Manager VPN Attachment (%s) delete: %s", d.Id(), err)
+	}
+
+	return nil
+}
+
+func FindVpnAttachmentByID(ctx context.Context, conn *networkmanager.NetworkManager, id string) (*networkmanager.SiteToSiteVpnAttachment, error) {
+	input := &networkmanager.GetSiteToSiteVpnAttachmentInput{
+		AttachmentId: aws.String(id),
+	}
+
+	output, err := conn.GetSiteToSiteVpnAttachmentWithContext(ctx, input)
+
+	if tfawserr.ErrCodeEquals(err, networkmanager.ErrCodeResourceNotFoundException) {
+		return nil, &resource.NotFoundError{
+			LastError:   err,
+			LastRequest: input,
+		}
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	if output == nil || output.SiteToSiteVpnAttachment == nil || output.SiteToSiteVpnAttachment.Attachment == nil {
+		return nil, tfresource.NewEmptyResultError(input)
+	}
+
+	return output.SiteToSiteVpnAttachment, nil
+}
+
+func StatusVpnAttachmentState(ctx context.Context, conn *networkmanager.NetworkManager, id string) resource.StateRefreshFunc {
```

Comment:
This doesn't need to be exposed outside of the package

```suggestion
func statusVpnAttachmentState(ctx context.Context, conn *networkmanager.NetworkManager, id string) resource.StateRefreshFunc {
```

Reply:
Changed to private